### ISSUE TITLE
Transactional templates substitution tags in one-mail-to-many-users use case need to be looped.

### ISSUE THREAD

**Issue by h5necom** [sendgrid/sendgrid-csharp#224]
Hi,

We are sending a same notification mail to many users. 
The mail uses templating engine and has substitution tags for amount of data transferred etc.
The substitution tags have exactly same information for each mail receiver.

Now in this case in code the following loop needs to be done, where the substitutionValues list given to AddSubstitution needs to be filled with same value for as many receivers there are, which is a bit awkward. 
(Otherwise only first receiver would have substituted values in place if given only first value)

public virtual ActionResult TrafficLimitExceeded(IList<string> emails, TrafficLimitExceeded model)
        {
            var emailMessage = new SendGridMessage();
            emailMessage.From = new MailAddress(EmailFrom, "Support Team");
            emailMessage.AddTo(emails);
            emailMessage.Subject = "Traffic limit exceeded";
            emailMessage.EnableTemplateEngine(trafficLimitExceededTemplateId);
            emailMessage.AddSubstitution("%%request_size_limit%%", GetRepeatedList(emails.Count,     Convert.ToString(model.RequestSizeLimitMb)));
...
}

private List<String> GetRepeatedList(int count, String entry)
        {
            List<String> repeatedList = new List<String>();
            for (int i = 0; i < count; i++) { repeatedList.Add(entry); }
            return repeatedList;
        }

I was hoping the AddSubstitution method could have an overload to take a single String parameter instead List <String>, and then the single value would be used for all receivers. 
Maybe already in v3 api? :) I think it would be more user friendly, it took some time to ponder how the substitution value list worked, as i had this kind of use case to implement as first.

**Comment by thinkingserious** [comment]
Hello @h5necom,

Thank you for taking the time to reach out to us.

I don't have a quick answer for you at the moment; however, you can check out the new v3 /mail/send endpoint here: https://sendgrid.com/docs/API_Reference/Web_API_v3/Mail/index.html and the supporting library here: https://github.com/sendgrid/sendgrid-csharp/tree/v3beta

I've tagged this issue for follow up in the mean time.

With Best Regards,

Elmer

**Comment by thinkingserious** [comment]
Hello @h5necom,

Have you upgraded to v7.0.3 yet? Did you get a chance to take a look at the v3 mail/send docs?

**Comment by h5necom** [comment]
I took a brief look,

https://github.com/sendgrid/sendgrid-csharp/tree/v3beta gives 404 not found atm,
i guess was removed already to master?

https://github.com/sendgrid/sendgrid-csharp/blob/master/SendGrid/SendGrid/Helpers/Mail/Mail.cs
Now method signature is public void AddSubstitution(String key, String value) instead of a substitution tag name and List parameters.

Ah, i see now that AddTo(Email email) method takes now only a single Email instead of a list of Emails, so I guess this got "fixed", because api user can now loop and create a new message for each recipient with same substitution tags using the Personalization features, which makes more sense.

Will try out at some point and see if i encounter any issues.
